### PR TITLE
feat: pass nix flake args to environment subcommands

### PR DESCRIPTION
**Pull request by limeytexan**
## Proposed Changes

Enable parsing and passing of nix flake related options to flox environment-related commands. In particular this facilitates the testing of flox-floxpkgs library changes with:

$ flox install foo --override-input flox-floxpkgs ~/wherever

## Checks

- [x] All tests pass.
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] I have accepted the flox [Contributor License Agreement](../blob/main/.github/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](../blob/main/.github/CONTRIBUTORS.csv) file by way of this pull request or one done previously.

## Release Notes

flox subcommands updated to accept flake-related options for the building of flox environment packages.

### CODE REVIEW

File: flox-bash/lib/commands.sh [hunk 1]
```diff
@@ -108,6 +108,43 @@ function parseNixArgs() {
...
+			;;
+		# Options taking one arg.
+		--priority|--eval-store|--include|-I|--inputs-from|--update-input|--expr|--file|-f)
+			_floxFlakeArgs+=("$1"); shift
+			_floxFlakeArgs+=("$1"); shift
+			;;
+		# Options taking zero args.
+		--debugger|--impure|--commit-lock-file|--no-registries|--no-update-lock-file|--no-write-lock-file|--recreate-lock-file|--derivation)
+			_floxFlakeArgs+=("$1"); shift
+			;;
```
Comment (limeytexan) [inline flox-bash/lib/commands.sh:134]: @ysndr I took these lines from the previous implementation of `floxInstall()`; some of these may not be accepted/required by other subcommands which end up building flox environments. We should meet offline to decide which ones we no longer require (if any), and discuss a strategy for updating the rust end to pass them through.

**Comment by ghudgins** [comment]
@limeytexan for my education why do we need this to test?

**Comment by limeytexan** [comment]
> @limeytexan for my education why do we need this to test?

@ghudgins in the process of performing the various environment manipulation subcommands flox will perform a `nix build` of the updated environment, and it isn't currently possible to pass options to this [embedded] invocation from the command line. One of the good reasons for wanting to pass options to this build is for debugging, and specifically the ability to override the `flox-floxpkgs` input that provides the bulk of the floxEnv-rendering Nix library functionality is _essential_ when debugging the floxEnv builder.

In our wrapping of the various nix "build-related" commands we've already arranged to pass the arguments unaltered, but the environment subcommands are of our own making, and we are therefore at liberty to treat those arguments however we like. This patch makes the assumption that _if_ you are passing nix "build" arguments to a flox environment-related subcommand then you probably want those arguments to be passed to the underlying `nix build` that is being invoked to build the new environment, and it's useful for the reasons noted above.

File: flox-bash/lib/commands.sh [hunk 2]
```diff
@@ -108,6 +108,43 @@ function parseNixArgs() {
...
+		# Consume remaining args
+		--command|-c|--)
+			_cmdArgs+=("$@")
+			break
+			;;
```
Comment (ysndr) [inline flox-bash/lib/commands.sh:139]: we got this twice now; also in https://github.com/flox/flox/pull/110/files#diff-f2b7203d41a3b7ded798568a5d9c980434118874c941694821bfa15599a2ffe2R99

File: flox-bash/lib/commands/development.sh [hunk 3]
```diff
@@ -243,8 +246,9 @@ _usage["print-dev-env"]="print shell code that can be sourced by bash to reprodu
 function floxDevelop() {
 	trace "$@"
 	parseNixArgs "$@" && set -- "${_cmdArgs[@]}"
+	parseFloxFlakeArgs "$@" && set -- "${_cmdArgs[@]}"
 
-	local -a developArgs=()
+	local -a developArgs=( "${_floxFlakeArgs[@]}" )
 	local -a installables=()
 	local -a remainingArgs=()
 	while test $# -gt 0; do
```
Comment (ysndr) [inline flox-bash/lib/commands/development.sh:254]: were parsing a subset of nix flake args here again, although refactored into _floxFlakeArgs

File: flox-bash/lib/commands/environment.sh [hunk 4]
```diff
@@ -55,9 +57,12 @@ function floxList() {
 	trace "$@"
 	local environment="$1"; shift
 	local system="$1"; shift
+	parseNixArgs "$@" && set -- "${_cmdArgs[@]}"
+	parseFloxFlakeArgs "$@" && set -- "${_cmdArgs[@]}"
 	# set $branchName,$floxNixDir,$environment{Name,Alias,Owner,System,BaseDir,BinDir,ParentDir,MetaDir}
 	eval $(decodeEnvironment "$environment")
 	parseNixArgs "$@" && set -- "${_cmdArgs[@]}"
+	parseFloxFlakeArgs "$@" && set -- "${_cmdArgs[@]}"
 	local -a invocation=("$@")
```
Comment (ysndr) [inline flox-bash/lib/commands/environment.sh:66]: doe we really need the parse*Args twice here? anything that decodeEnvironment would add to it?

File: flox-bash/lib/commands/development.sh (hunk 3 above)
Comment (ysndr) [reply flox-bash/lib/commands/development.sh:254]: also in other development commands

**Review by ysndr — CHANGES_REQUESTED** [review]

File: flox-bash/lib/commands/environment.sh (hunk 4 above)
Comment (limeytexan) [reply flox-bash/lib/commands/environment.sh:66]: Nope that's a mistake ... well spotted, will remove the second instance.

File: flox-bash/lib/commands.sh (hunk 2 above)
Comment (limeytexan) [reply flox-bash/lib/commands.sh:139]: This is correct; each time we call either of `parse{NixArgs,FloxFlakeArgs}()` we need to return by way of `${_cmdArgs[@]}` any args following a `--command` or `--` arg. Think of this as a passthru.

File: flox-bash/lib/commands/development.sh (hunk 3 above)
Comment (limeytexan) [reply flox-bash/lib/commands/development.sh:254]: Thanks for highlighting - pushed a commit to remove the arg parsing now handled by `parseFloxFlakeArgs()`.

**Review by ysndr — APPROVED** [review]

**Pull request merged**